Common/src/config_structure.cpp

```diff
@@ -4041,10 +4041,6 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
 
       Restart_Flow = false;
 
-      if (GetGrid_Movement()) {
-        SU2_MPI::Error("Dynamic mesh movement currently not supported for the discrete adjoint solver.", CURRENT_FUNCTION);
```

Comment:
I guess it only works for rigid movement at the moment then, am I correct ?

Reply:
Correct. Now checks` Kind_GridMovement != RIGID_MOTION` and throws an error.

Reply:
Rigid motion cases have been used for the verification at the moment but GCL is included to handle changing volumes of a mesh in the primal solvers. Therefore the changes in this request should capture both rigid motion and deforming mesh problems for discrete adjoints as the grid velocities are now captured correctly. Please advise if I'm missing to account for something else here? If anyone can point me to a good single zone mesh deformation test case that already exists for the primal solver, I can investigate the discrete adjoint explicitly for a deforming mesh problem as well.